### PR TITLE
pulled_your_commits_@praticr_check_for_changes 

### PR DESCRIPTION
**Pull request by sloppy-genius**

### CODE REVIEW

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/MainActivity.java [hunk 1]
```diff
@@ -73,12 +74,35 @@ public void onSuccess() {
...
+
+                    @Override
+                    public void onFailure() {
+                        failureAction.perform();
+                    }
+                }));
+    }
+
+    //ToDo : find a more proper implementation
+    public void signUpComplete() {
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/MainActivity.java]: rename to openCelebrityList

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 2]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
         return StringUtils.isNotEmpty(getUserId(context));
     }
 
-    public static void doWithAuthentication(final MainActivity mainActivity,
-                                            final Action successAction,
-                                            final Action failureAction) {
+    public static void doIfAuthenticated(final MainActivity mainActivity,
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: rename back

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 3]
```diff
@@ -20,6 +26,8 @@
     public static final String TAG = MainActivity.class.getSimpleName();
     public static final String LOG_TAG_BASE = "PAPPARAZI_";
     public static final String LOG_TAG = LOG_TAG_BASE + TAG;
+    //variable to return if the task was successful or not
+    public static boolean checkForSuccessful = false;
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: make local

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 4]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
...
+                                     final Action successAction) {
+
+        doIfNewUser(mainActivity, successAction, () ->
+                Toast.makeText(mainActivity,
+                        "SignUp Unsuccessful. Please Try Again", Toast.LENGTH_SHORT)
+                        .show());
+    }
+
+    //ToDo : find a better way to handle activity
+    public static boolean completeSignIn(String email, String password, Activity mainActivity) {
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: rename signIn

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 5]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
...
+
+        ExternalUtils.getFirebaseAuth().signInWithEmailAndPassword(email, password)
+                .addOnCompleteListener(mainActivity, task -> {
+
+                    checkForSuccessful = false;
+                    if (task.isSuccessful()) {
+
+                        // Sign in success, update UI with the signed-in user's information
+                        Log.d(LOG_TAG, "successfully signed in with email");
+                        Toast.makeText(mainActivity.getBaseContext(), "Successfully Logged In",
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: dont do ui related things in AuthUtils, do wherever necessary

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 6]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
...
+                                Toast.LENGTH_SHORT).show();
+                        setUserId(mainActivity.getBaseContext(),
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: do you need baseContext?

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 7]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
...
+                                    .getErrorCode()) {
+
+                                case "ERROR_USER_NOT_FOUND" : {
+                                    message = "Wrong E-mail Address. Please Try Again";
+                                    break;
+                                }
+                            }
+                        }
+
+                        Toast.makeText(mainActivity.getBaseContext(), message, Toast.LENGTH_SHORT).show();
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: make a response pojo

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 8]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
...
+
+                        //sign in user after successful sign up
+
+                        //ToDo : (developer)check for a proper solution
+                        //there could be a problem here if user gets registered and
+                        //sign in fails we are stuck and the user cannot create another login
+                        checkForSuccessful = completeSignIn(email, password, mainActivity);
+
+                        //ToDo : check if this method is necessary or not
+                        setUserId(mainActivity.getBaseContext(),
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: remove this

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/interfaces/OnFragmentActionPerformedListener.java [hunk 9]
```diff
@@ -7,7 +7,6 @@
  */
 
 public interface OnFragmentActionPerformedListener extends Serializable {
-
     void onSuccess();
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/interfaces/OnFragmentActionPerformedListener.java]: ??

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 10]
```diff
@@ -31,21 +28,20 @@
     public static final String LOG_TAG = LOG_TAG_BASE + TAG;
 
     public static SignInFragment newInstance(final OnFragmentActionPerformedListener
-                                                 actionPerformedListener) {
+                                                     actionPerformedListener) {
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: remove this

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 11]
```diff
@@ -31,21 +28,20 @@
...
 
         SignInFragment signInFragment = new SignInFragment();
         signInFragment.setArguments(new Bundle());
         signInFragment.getArguments().putSerializable(Argument.ACTION_PERFORMED.key,
-            actionPerformedListener);
+                actionPerformedListener);
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: remove this

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 12]
```diff
@@ -31,21 +28,20 @@
...
         return signInFragment;
     }
 
     public static SignInFragment newInstanceIfRequired(final MainActivity mainActivity,
                                                        final OnFragmentActionPerformedListener
                                                                actionPerformedListener) {
-
         return ObjectUtils.defaultIfNull((SignInFragment) mainActivity
-            .getSupportFragmentManager().findFragmentByTag(SignInFragment.TAG),
+                .getSupportFragmentManager().findFragmentByTag(SignInFragment.TAG),
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: remove this

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 13]
```diff
@@ -86,7 +83,7 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         }
 
         onFragmentActionPerformedListener = (OnFragmentActionPerformedListener) getArguments()
-            .getSerializable(Argument.ACTION_PERFORMED.key);
+                .getSerializable(Argument.ACTION_PERFORMED.key);
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: remove this

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 14]
```diff
@@ -157,59 +162,15 @@ private void validateSignInDetails() {
...
+            removeFragment();
+        } else {
+            onFragmentActionPerformedListener.onFailure();
+        }
 
-                Toast.makeText(getContext(), message, Toast.LENGTH_SHORT).show();
-            }
-        });
     }
+
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: remove this

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java [hunk 15]
```diff
@@ -48,25 +56,156 @@ public static boolean isUserAuthenticated(final Context context) {
...
     public static void doWithAuthentication(final MainActivity mainActivity,
-                                            final Action successAction) {
+                                       final Action successAction) {
+
+        doIfAuthenticated(mainActivity, successAction, () ->
+                Toast.makeText(mainActivity,
+                        "SignIn Unsuccessful. Please Try Again", Toast.LENGTH_SHORT)
+                        .show());
+    }
+    public static void doIfNewUser(final MainActivity mainActivity,
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/commons/AuthUtils.java]: do with new auth

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/MainActivity.java (hunk 1 above)
Comment (sloppy-genius) [reply app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/MainActivity.java]: done

**Pull request closed without merging**